tests/test_frame.py

```diff
@@ -1780,16 +1779,15 @@ def test_set_columns() -> None:
     # https://github.com/python/mypy/issues/3004
     # pyright accepts this, so we only type check for pyright,
     # and also test the code with pytest
-    if (TYPE_CHECKING and not IS_TYPE_CHECKER_MYPY) or not TYPE_CHECKING:
-        df.columns = ["c", "d"]
-        df.columns = [1, 2]
-        df.columns = [1, "a"]
-        df.columns = np.array([1, 2])
-        df.columns = pd.Series([1, 2])
-        df.columns = np.array([1, "a"])
-        df.columns = pd.Series([1, "a"])
-        df.columns = (1, 2)
-        df.columns = (1, "a")
+    df.columns = ["c", "d"]  # type: ignore[assignment]
+    df.columns = [1, 2]  # type: ignore[assignment]
+    df.columns = [1, "a"]  # type: ignore[assignment]
+    df.columns = np.array([1, 2])  # type: ignore[assignment]
+    df.columns = pd.Series([1, 2])  # type: ignore[assignment]
+    df.columns = np.array([1, "a"])  # type: ignore[assignment]
+    df.columns = pd.Series([1, "a"])  # type: ignore[assignment]
+    df.columns = (1, 2)  # type: ignore[assignment]
+    df.columns = (1, "a")  # type: ignore[assignment]
```

Comment:
Can you add this kind of test here?
```python
        if TYPE_CHECKING_INVALID_USAGE:
            df.columns = "abc"  # pyright: ignore[reportGeneralTypeIssues]
```
Want to make sure this works

Reply:
You added that in the other PR, or?

Reply:
Yes.  So with my changes there, you'll have to update this PR to use the correct `#pyright ignore` or `#type: ignore` as needed.

Reply:
Should be ready now